src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,67 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { User } from '../../types/User';
+import { useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { useDispatch } from 'react-redux';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getUser(currentTodo?.userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
```

Comment:
Consider adding error handling for the `getUser` API call. If the API call fails, the user will be stuck in a loading state. You can add a `.catch` block to handle any errors and provide feedback to the user.